core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -382,71 +383,57 @@ public List<TopicRequest> selectTopicRequestsByStatus(
-                .collect(Collectors.toList());
-
-        claimTopicReqs =
-            claimTopicReqs.stream()
-                .filter(
-                    claimTopicReq ->
-                        Objects.equals(claimTopicReq.getDescription(), "" + teamSelected)
-                            && Objects.equals(claimTopicReq.getTopicstatus(), status))
-                .collect(Collectors.toList());
-      }
-
-      topicRequestListSub.addAll(claimTopicReqs);
+    if (showRequestsOfAllTeams) {
+      topicRequests = topicRequestsRepo.findAllByTenantId(tenantId);
     } else {
-      topicRequestListSub = Lists.newArrayList(topicRequestsRepo.findAllByTenantId(tenantId));
+      topicRequests = topicRequestsRepo.findAllByTeamIdAndTenantId(teamSelected, tenantId);
     }
 
-    for (TopicRequest row : topicRequestListSub) {
-      Integer teamName = row.getTeamId();
-      try {
-        row.setRequesttimestring(
-            (new SimpleDateFormat("dd-MMM-yyyy HH:mm:ss"))
-                .format((row.getRequesttime()).getTime()));
-      } catch (Exception ignored) {
-      }
+    if (allReqs) { // approvers
 
-      if (showRequestsOfAllTeams) {
-        topicRequestList.add(row); // no team filter
-      } else if (teamSelected != null
-          && (teamSelected.equals(teamName) || ("" + teamSelected).equals(row.getDescription()))) {
-        topicRequestList.add(row);
-      }
+      List<TopicRequest> claimTopicReqs =
+          topicRequestsRepo.findAllByTopictypeAndTenantId(
+              RequestOperationType.CLAIM.value, tenantId);
+      topicRequests =
+          topicRequests.stream()
+              .filter(
+                  topicRequest ->
+                      !RequestOperationType.CLAIM.value.equals(topicRequest.getTopictype()))
+              .collect(Collectors.toList());
+      claimTopicReqs =
+          claimTopicReqs.stream()
+              .filter(
+                  claimTopicReq ->
+                      Objects.equals(claimTopicReq.getDescription(), "" + teamSelected))
+              .collect(Collectors.toList());
+
+      topicRequests.addAll(claimTopicReqs);
+
+      // remove users own requests to approve/show in the list
+      topicRequests =
+          topicRequests.stream()
+              .filter(topicRequest -> !topicRequest.getRequestor().equals(requestor))
+              .collect(Collectors.toList());
     }
 
-    return topicRequestList;
+    if (!"all".equals(status)) {
+      topicRequests =
+          topicRequests.stream()
+              .filter(topicRequest -> topicRequest.getTopicstatus().equals(status))
+              .collect(Collectors.toList());
+    }
+
+    topicRequests.forEach(
+        topicRequest ->
+            topicRequest.setRequesttimestring(
+                new SimpleDateFormat("dd-MMM-yyyy HH:mm:ss")
+                    .format((topicRequest.getRequesttime()).getTime())));
+
+    return topicRequests;
   }
```

Comment:
This will be all changed in my upcoming PR. 
But I will need to make sure it matches this logic correctly as I believe this is a bugfix related to what Mathieu discussed with us earlier. (More of an FYI comment.)